src/connector/src/parser/debezium/simd_json_parser.rs

```diff
@@ -298,7 +298,7 @@ mod tests {
                 SourceColumnDesc::simple("O_DATE", DataType::Date, ColumnId::from(8)),
                 SourceColumnDesc::simple("O_TIME", DataType::Time, ColumnId::from(9)),
                 SourceColumnDesc::simple("O_DATETIME", DataType::Timestamp, ColumnId::from(10)),
-                SourceColumnDesc::simple("O_TIMESTAMP", DataType::Timestamp, ColumnId::from(11)),
+                SourceColumnDesc::simple("O_TIMESTAMP", DataType::Timestamptz, ColumnId::from(11)),
```

Comment:
When tightening the parsing of naive `timestamp`, I also found this typing issue in debezium mysql source: `timestamp` in MySQL should be mapped to `timestamptz` in PostgreSQL. Are there any other places I need to fix in addition to the tests here? @StrikeW

Reply:
I will add a validate check in mysql-cdc to enforce `timestamp` map to `timestampz` after this pr merged.

Reply:
Would you mind taking a look at this part? Not sure whether we need to modify here also.
https://github.com/risingwavelabs/risingwave/blob/9ccdbe368ace5a3ada11a8c82952b24a0104f6d5/src/connector/src/parser/mysql.rs#L74-L81

Reply:
Thanks for the info!
* For mysql-cdc validation, it may be easier to be in the same PR if we want to cherry pick it into 1.2.x. Alternatively, I can also make this PR less defensive to not reject such typing mistake during parser until 1.3. But we add the checking in mysql-cdc in 1.2 first.
* I can confirm this `mysql_async` mapping need no change, because the data for `datetime` and `timestamp` look exactly the same. This is different from [debezium](https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-temporal-types:~:text=DATETIME%20with%20a,37%3A03Z.). However, we also lose the opportunity to do validation as in the debezium case. Furthermore the mysql client should be configured to set session timezone to utc. IIRC this is already done.